bazaarvoice/emodb
# EMO-6491: Added ability to create custom roles on start up from web-local

## Github Issue #

None

## What Are We Doing Here?

When the default "anonymous" permissions were removed from Emo this had the side effect of removing all anonymous permissions when starting Emo using the scripts from web-local.  While use of anonymous access is discouraged for those clients who rely on it this left them with no simple way to set those permissions on start up like the SDK allows.

This PR introduces the ability to specify a permissions file when running `web-local/start.sh`.  The script will generate all roles in the file once Emo is started automatically.  It also includes a file `anonymous-permissions.yaml` which contains the old anonymous permissions which used to be included in Emo by default.

## How to Test and Verify

1. Check out this PR
2. Build Emo
3. From `web-local` run `start.sh`
4. Call `curl -v localhost:8080/sor/1/foo` and verify the response is 403.
5. Kill Emo using ctrl-c.
6. From `web-local` run `start.sh --permissions-file anonymous-permissions.yaml`
7. Call `curl -v localhost:8080/sor/1/foo` and verify the response is 404 (anonymous now has permission to read the table, although the table does not exist).

## Risk

This is a low risk update.  It only affects users running Emo locally using `web-local`.  No changes to the application itself were made.

### Level 

`Low`

### Required Testing

`Manual`

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
